compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -560,4 +629,342 @@ private BLangLiteral updateLiteral(BLangLiteral literal, Object value, BType typ
+        origTypeRef.pos = pos;
+        origTypeRef.setBType(origStructureType);
+
+        ((BLangRecordTypeNode) recordTypeNode).sealed = true;
+    }
+
+    private BLangSimpleVariable createSimpleVariable(BField field) {
+        BLangSimpleVariable manualField = new BLangSimpleVariable();
+        BLangIdentifier name = new BLangIdentifier();
+
+        name.setValue(field.name.value);
+        name.pos = field.pos;
+        manualField.setName(name);
+
+        manualField.flagSet = new HashSet<>();
+        manualField.flagSet.add(Flag.PUBLIC);
+        manualField.flagSet.add(Flag.REQUIRED);
+        manualField.flagSet.add(Flag.FIELD);
+        manualField.setBType(field.type);
+        manualField.pos = field.pos;
+        manualField.setDeterminedType(field.type);
+        manualField.symbol = field.symbol;
+
+        if (field.type.tag == TypeTags.RECORD) {
+            BLangUserDefinedType userDefinedTypeNode = (BLangUserDefinedType) TreeBuilder.createUserDefinedTypeNode();
+            userDefinedTypeNode.pos = field.pos;
+            userDefinedTypeNode.setBType(field.type);
+            userDefinedTypeNode.setDeterminedType(field.type);
+            userDefinedTypeNode.pkgAlias = new BLangIdentifier();
+            if (createdTypeDefinitions.containsKey(field.type.tsymbol)) {
+                BLangTypeDefinition createdType = createdTypeDefinitions.get(field.type.tsymbol);
+                userDefinedTypeNode.symbol = createdType.symbol;
+                userDefinedTypeNode.typeName = createdType.name;
+            }
+
+            manualField.typeNode = userDefinedTypeNode;
+            return manualField;
+        }
+
+        BLangFiniteTypeNode finiteTypeNode = (BLangFiniteTypeNode) TreeBuilder.createFiniteTypeNode();
+        finiteTypeNode.pos = field.pos;
+        finiteTypeNode.setBType(field.type);
+        finiteTypeNode.setDeterminedType(field.type);
+        finiteTypeNode.valueSpace = new ArrayList<>();
+        for (BLangExpression value : ((BFiniteType) field.type).getValueSpace()) {
+            if (value.getKind() == NodeKind.NUMERIC_LITERAL) {
+                ((BLangNumericLiteral) value).originalValue = ((BLangNumericLiteral) value).value.toString();
+            }
+            finiteTypeNode.valueSpace.add(value);
+        }
+        manualField.typeNode = finiteTypeNode;
+        return manualField;
+    }
+
+    private BLangTypeDefinition findTypeDefinition(List<BLangTypeDefinition> typeDefinitionArrayList, String name) {
+        for (int i = typeDefinitionArrayList.size() - 1; i >= 0; i--) {
+            BLangTypeDefinition typeDefinition = typeDefinitionArrayList.get(i);
+            if (typeDefinition.name.value.equals(name)) {
+                return typeDefinition;
+            }
+        }
+        return null;
+    }
+
+    private BIntersectionType cloneTypeDefinition(BLangConstant constant, BConstantSymbol symbol,
```

Comment:
Why do we need this?

Reply:
This was used to create a new type definition using constant name (in addition to anonymous type def). Since we cannot filter out constant symbol and its definition separately in `lookupMemberSymbol` logic, we have removed this method.